File: replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) 
    {
        int n = arr.size();
        int max_so_far = -1;
        for(int i=n-1; i>=0; i--) 
        {
            int curr = arr[i];
            arr[i] = max_so_far;

            if(curr > max_so_far) 
            {
                max_so_far = curr;
            }
        }
        return arr;
    }
};

File: replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        max_so_far = -1
        for i in range(n-1, -1, -1):
            curr = arr[i]
            arr[i] = max_so_far
            if curr > max_so_far:
                max_so_far = curr
        return arr